app/src/vuex/modules/connection.js

```diff
@@ -151,7 +151,7 @@ export default function ({ apollo }) {
       commit("setNetworks", data.networks)
     },
     async setNetwork({ commit, dispatch }, network) {
-      dispatch(`signOut`)
+      dispatch(`signOut`, { address: undefined, networkId: network.id })
```

Comment:
why do we need to pass the address here? there should only be one address per network active

Reply:
because now I am storing all the sessions that an user ever had under `session_<networkId>_<address>`. This is now necessary since I need to make sure every time an account is selected that it logs in with the right algo

Reply:
the algo should not go into the session but into the wallet, so into cosmos-keys.

Reply:
So then we need to change cosmos-keys in the end. OK

Reply:
https://github.com/luniehq/cosmos-keys/pull/86